tools/releng/ls-missing-release-builds

```diff
@@ -84,25 +84,48 @@ if [[ -z "$JR_MANIFESTS" ]]; then
     JR_MANIFESTS=/Volumes/projects/triton.git/tools/jr-manifest.json,/Volumes/projects/manta.git/tools/jr-manifest.json,/Volumes/projects/smartos-live.git/tools/jr-manifest.json"
 fi
 
-targets="$*"
+filtered_targets="$*"
 if [[ -z "$targets" ]]; then
-    targets=$(jr list -H -l mg,release -o labels | json -ag mg buildisprivate | sort -u \
-        | grep -v '^sdcsso')
+    targets=$(jr list -H -l mg,release \
+        -o labels.mg,labels.buildisprivate,htmlUrl | grep -v ^sdcsso)
 fi
 
 # Ensure mls is setup properly at all.
 mls /Joyent_Dev/public/builds >/dev/null \
     || fatal "cannot list /Joyent_Dev/public/builds"
 
-echo "$targets" | while read target buildisprivate; do
+echo "$targets" | while read target buildisprivate htmlurl; do
+
+    # if we've been asked to only look for specific targets, then
+    # we should skip to the next item if this target isn't in our list
+    if [[ -n "$filtered_targets" ]]; then
+        process_target=""
+        for filter in $filtered_targets; do
+            if [[ "$target" == "$filter" ]]; then
+                process_target=true
+            fi
+        done
+        if [[ -z "$process_target" ]]; then
+            continue
+        fi
+    fi
+
     if [[ "$buildisprivate" == "true" ]]; then
         latest_mpath=/Joyent_Dev/stor/builds/$target/$release-latest
     else
         latest_mpath=/Joyent_Dev/public/builds/$target/$release-latest
     fi
 
+    gitrepo=$(echo $htmlurl | awk -F/ '{print $NF}')
+
     case $target in
-        "platform"|"platform-debug")
+        "platform")
+            # For the platform build, only the 'smartos-live' repository
+            # is relevant, ignore the others.
+            smartos_live=$(echo $gitrepo | grep smartos-live || true)
+            if [[ -z "$smartos_live" ]]; then
```

Comment:
Could be `if [[ "$gitrepo" == "smartos-live" ]]; then` as well, right? (I'm not sure if I'm missing something.)

Reply:
duh, thanks.